app/server/app/routes/data.js

```diff
@@ -1,11 +1,153 @@
+    query.whereIn(paramName, parsedParam.value);
+  } else {
+    query.where(paramName, parsedParam.value);
+  }
+}
+
+function parseCriteria(query, profile, queryParams) {
+  switch (profile) {
+    case "assessments":
+      appendToWhere(query, "id", queryParams.id);
+      appendToWhere(query, "reporting_cycle", queryParams.reportingCycle);
+      appendToWhere(query, "assessment_unit_id", queryParams.assessmentUnitId);
+      appendToWhere(
+        query,
+        "assessment_unit_name",
+        queryParams.assessmentUnitName
+      );
+      appendToWhere(query, "organization_id", queryParams.organizationId);
+      appendToWhere(query, "organization_name", queryParams.organizationName);
+      appendToWhere(query, "organization_type", queryParams.organizationType);
+      appendToWhere(query, "overall_status", queryParams.overallStatus);
+      appendToWhere(query, "region", queryParams.region);
+      appendToWhere(query, "state", queryParams.state);
+      appendToWhere(query, "ir_category", queryParams.irCategory);
+      break;
+    case "profile_test":
+      appendToWhere(query, "id", queryParams.id);
+      appendToWhere(query, "assessment_name", queryParams.assessmentName);
+      break;
+    default:
+      break;
+  }
+}
+
+async function executeQuery(profile, req, res, next) {
+  // output types csv, tab-separated, Excel, or JSON
+  try {
+    const query = knex.withSchema(req.activeSchema).select("*").from(profile);
+
+    parseCriteria(query, profile, req.query);
+
+    const stream = await query.stream({
+      batchSize: 2000,
+      highWaterMark: 10000,
+    });
+
+    const format = req.query.format ?? req.query.f;
+    switch (format) {
+      case "csv":
+      case "tsv":
+        // output the data
+        res.setHeader(
+          "Content-disposition",
+          `attachment; filename=${profile}.${format}`
+        );
+        StreamingService.streamResponse(res, stream, format);
+        break;
+      case "xlsx":
+        res.setHeader(
+          "Content-Disposition",
+          `attachment; filename=${profile}.xlsx`
+        );
+
+        const workbook = new Excel.stream.xlsx.WorkbookWriter({
+          stream: res,
+          useStyles: true,
+        });
+
+        workbook.addWorksheet(profile);
+        const worksheet = workbook.getWorksheet(profile);
+
+        stream
+          .pipe(
+            new ExcelTransform({
+              workbook,
+              worksheet,
+            })
+          )
+          .pipe(process.stdout);
+        break;
+      case "json":
+      default:
+        res.setHeader(
+          "Content-disposition",
+          `attachment; filename=${profile}.json`
+        );
+        StreamingService.streamResponse(res, stream, format);
+        break;
+    }
+  } catch (error) {
+    log.error(`Failed to get data from the "${profile}" profile...`);
+    return res.status(500).send("Error !" + error);
+  }
+}
+
+function executeQueryCountOnly(profile, req, res, next) {
```

Comment:
Same as above, the `next` parameter is unused:
```suggestion
function executeQueryCountOnly(profile, req, res) {
```